tests/Bridge/Symfony/Bundle/ArgumentResolver/PayloadArgumentResolverTest.php

```diff
@@ -0,0 +1,343 @@
+        $argument = $this->createArgumentMetadata(ResourceInterface::class);
+
+        $this->expectException(\InvalidArgumentException::class);
+
+        iterator_to_array($resolver->resolve($request, $argument));
+    }
+
+    public function provideUnsupportedRequests(): iterable
+    {
+        yield 'GET request' => [$this->createRequest('GET', [
+            '_api_resource_class' => ResourceImplementation::class,
+            '_api_receive' => true,
+            '_api_item_operation_name' => 'update',
+            'data' => new ResourceImplementation(),
+        ])];
+
+        yield 'HEAD request' => [$this->createRequest('HEAD', [
+            '_api_resource_class' => ResourceImplementation::class,
+            '_api_receive' => true,
+            '_api_item_operation_name' => 'update',
+            'data' => new ResourceImplementation(),
+        ])];
+
+        yield 'OPTIONS request' => [$this->createRequest('OPTIONS', [
+            '_api_resource_class' => ResourceImplementation::class,
+            '_api_receive' => true,
+            '_api_item_operation_name' => 'update',
+            'data' => new ResourceImplementation(),
+        ])];
+
+        yield 'TRACE request' => [$this->createRequest('TRACE', [
+            '_api_resource_class' => ResourceImplementation::class,
+            '_api_receive' => true,
+            '_api_item_operation_name' => 'update',
+            'data' => new ResourceImplementation(),
+        ])];
+
+        yield 'DELETE request' => [$this->createRequest('DELETE', [
+            '_api_resource_class' => ResourceImplementation::class,
+            '_api_receive' => true,
+            '_api_item_operation_name' => 'update',
+            'data' => new ResourceImplementation(),
+        ])];
+
+        yield 'request without attributes' => [$this->createRequest('PUT', [])];
+
+        yield 'request with receive=false' => [$this->createRequest('PUT', [
+            '_api_resource_class' => ResourceImplementation::class,
+            '_api_receive' => false,
+            '_api_item_operation_name' => 'update',
+            'data' => new ResourceImplementation(),
+        ])];
+
+        yield 'request on operation with deserialization disabled' => [$this->createRequest('PUT', [
+            '_api_resource_class' => ResourceImplementation::class,
+            '_api_receive' => true,
+            '_api_item_operation_name' => 'update_no_deserialize',
+            'data' => new ResourceImplementation(),
+        ])];
+    }
+
+    public function provideUnsupportedArguments(): iterable
+    {
+        yield 'argument without type declaration' => [$this->createArgumentMetadata()];
+        yield 'variadic argument' => [$this->createArgumentMetadata(ResourceImplementation::class, true)];
+    }
+
+    /**
+     * @dataProvider provideIntegrationCases
+     */
+    public function testIntegration(Request $request, callable $controller, array $expectedArguments): void
```

Comment:
Shouldn't we add the integration tests in another file? If possible we prefer to use Behat for this kind of tests too (but maybe it would be harder to test it and these tests seem OK to maintain).

Reply:
> If possible we prefer to use Behat

I think it would be much harder to have reliable Behat tests and would require more work to write and maintain. Not worth it IMO.

> Shouldn't we add the integration tests in another file

IMO it's fine to keep it here: all tests in this file cover the same subject. But if you prefer, I can extract it to it own file.